crates/sargon/src/gateway_api/methods/state_methods.rs

```diff
@@ -86,3 +86,53 @@ impl GatewayClient {
             .await
     }
 }
+
+impl GatewayClient {
+    /// Fetches all the account's resource preferences.
+    pub async fn fetch_all_account_resource_preferences(
+        &self,
+        account_address: AccountAddress,
+        at_ledger_state: LedgerStateSelector,
+    ) -> Result<Vec<AccountResourcePreference>> {
+        self.load_all_pages(|response_cursor, response_ledger_state| {
+            let selector =
```

Comment:
If you allow passing in the intial selector to load_all_pages, you don't need this